mautic/mautic
# Fix/change hubspot api auth

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✅ 
| New feature/enhancement? (use the a.x branch)      | [
| Deprecations?                          | 
| BC breaks? (use the c.x branch)        | 
| Automated tests included?              | 
| Related user documentation PR URL      | https://github.com/mautic/mautic-documentation/pull/333
| Related developer documentation PR URL | 
| Issue(s) addressed                     | Fixes #... 



#### Description:
We need to change authentication method for Hubspot plugin. **New installation cannot use our plugin anymore**, and existing in integration will shut down in November, 22'.

Source: [developers.hubspot.com/changelog/upcoming-api-key-sunset](https://developers.hubspot.com/changelog/upcoming-api-key-sunset)

> What’s changing?
> API Keys have been [one of three authentication methods supported by HubSpot](https://developers.hubspot.com/docs/api/intro-to-auth) APIs. However, as part of ongoing efforts to protect our customer's data, we will be sunsetting API Keys.
> 
> As a result of this change, integrations will instead be required to work with [Private Apps](https://developers.hubspot.com/docs/api/private-apps). Private Apps offer tighter security and allow more granular control over your integrations and account data than legacy API keys.
> 
> What this means for developers:
> With the introduction of Private Apps, users who previously developed on HubSpot and utilized API Keys will now be required to migrate existing integrations from using API Key authentication to using Private Apps instead. Instructions for how to migrate existing integrations can be found [here](https://developers.hubspot.com/docs/api/migrate-an-api-key-integration-to-a-private-app).

Migration doc: [developers.hubspot.com/docs/api/migrate-an-api-key-integration-to-a-private-app](https://developers.hubspot.com/docs/api/migrate-an-api-key-integration-to-a-private-app)

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a private app in hubspot with following rights
3. Setup plugin integration
4. See everything works
![image](https://user-images.githubusercontent.com/14075239/189848029-4edceb9a-2c5a-418a-8e15-00b7725a0a28.png)
